pkg/discovery/discovery.go

```diff
@@ -255,16 +306,35 @@ func getPodLogNamespaceFilter(cfg *config.Config) string {
 }
 
 // updateStatus changes the summary status of the sonobuoy pod in order to
-// effect the finalized status the user sees. This does not change the status
-// of individual plugins.
-func updateStatus(client kubernetes.Interface, namespace string, status string) error {
+// effect the finalized status the user sees. This does not change the
+// status of individual plugins.
+func updateStatus(client kubernetes.Interface, namespace string, status string, tarInfo *pluginaggregation.TarInfo) error {
 	podStatus, err := pluginaggregation.GetStatus(client, namespace)
 	if err != nil {
 		return errors.Wrap(err, "failed to get the existing status")
 	}
 
 	// Update status
 	podStatus.Status = status
+	if tarInfo != nil {
+		podStatus.Tarball = *tarInfo
+	}
+	return setStatus(client, namespace, podStatus)
+}
+
+func updatePluginStatus(client kubernetes.Interface, namespace string, pluginType string, pluginResultStatus string, pluginResultCounts map[string]int) error {
+	podStatus, err := pluginaggregation.GetStatus(client, namespace)
+	if err != nil {
+		return errors.Wrap(err, "failed to get the existing status")
+	}
+
+	for i := range podStatus.Plugins {
+		if podStatus.Plugins[i].Plugin == pluginType {
```

Comment:
Is the plugin result type unique per run? I think I remember seeing that it was but it's slightly confusing to me here because `podStatus.Plugins[i].Plugin` seems like it would be the plugin's name. I know that our built in plugins use the same values for the name and the result type though.

Reply:
Yes the type is unique; agree that name/type duality is confusing and I want to fix that. See #830 ; someone reported an issue which was caused because of that too.